src/components/Mapbox.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useRef, useEffect } from 'react';
+import mapboxgl from 'mapbox-gl';
+import { Map, Marker, MapboxOptions, MarkerOptions, LngLatBounds } from 'mapbox-gl';
+import { Result, useSearchState } from '@yext/search-headless-react';
+import 'mapbox-gl/dist/mapbox-gl.css';
+import ReactDOM from 'react-dom';
+
+/**
+ * Interface for customizing the {@link Mapbox} map derived from Mapbox API's Map options.
+ * The "container" field is set internally. The "center" field is offered as a surface level prop
+ * in {@link MapboxProps}.
+ *
+ * @public
+ */
+export type MapboxCustomOptions = Omit<MapboxOptions, 'container' | 'center'>;
+
+/**
+ * Interface for customizing the {@link Mapbox} marker derived from Mapbox API's Marker options.
+ * The "element" field is set internally.
+ *
+ * @public
+ */
+export type MarkerCustomOptions = Omit<MarkerOptions, 'element'>;
+
+/**
+ * An outline to create a custom Marker component.
+ *
+ * @public
+ */
+export type MapPinComponent<TEntity> = (props: { result: Result<TEntity> }) => JSX.Element;
+
+/**
+ * Props for the {@link Mapbox} component.
+ *
+ * @public
+ */
+export interface MapboxProps<TEntity> {
+  mapboxApiKey: string,
+  mapboxOptions?: MapboxCustomOptions,
+  markerOptions?: MarkerCustomOptions,
+  PinComponent?: MapPinComponent<TEntity>,
+  defaultCenter?: Coordinate,
+  getCoordinate: CoordinateGetter<TEntity>
+}
+
+/**
+ * A component that renders a map with markers to show result locations.
+ *
+ * @param props - {@link MapboxProps}
+ * @returns A React element conatining a Google Map
+ *
+ * @public
+ */
+export function Mapbox<TEntity>({
+  mapboxApiKey,
+  mapboxOptions,
+  markerOptions,
+  PinComponent,
+  defaultCenter: center,
+  getCoordinate
+}: MapboxProps<TEntity>) {
+  mapboxgl.accessToken = mapboxApiKey;
+  const mapContainer = useRef(null);
+  const map = useRef<Map | null>(null);
+  const markers = useRef<Marker[]>([]);
+
+  const locationResults = useSearchState(state => state.vertical.results) as Result<TEntity>[];
+  const hasLocationResults = locationResults && locationResults.length !== 0;
+
+  const mapboxStyle = mapboxOptions?.style ?? 'mapbox://styles/mapbox/streets-v11';
+
+  useEffect(() => {
```

Comment:
should these effects have dependency arrays, or do they really need to run every single render?

Reply:
ahh no it definitely does not need to run every render